### PR TITLE
feat: add Room DAO, Data, Repository, SpringRepository, ControllerTest

### PR DESCRIPTION
**Pull request by cool9850311**

### CODE REVIEW

File: spring/src/main/kotlin/tw/waterballsa/gaas/spring/repositories/data/RoomData.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    var gameId: String?,
+    var hostId: String?,
+    var playerIds: List<String>?,
+    var maxPlayer: Int?,
+    var minPlayer: Int?,
+    var name: String?,
+    var description: String?,
+) {
+    companion object {
+        fun toData(room: Room): RoomData {
```
Comment (frankvicky) [inline spring/src/main/kotlin/tw/waterballsa/gaas/spring/repositories/data/RoomData.kt:21]: 改為 `fun Room.toData(): RoomData`

File: application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package tw.waterballsa.gaas.application.repositories
+
+import tw.waterballsa.gaas.domain.Room
+
+interface RoomRepository {
+    fun findByRoomId(roomId: String): Room?
```
Comment (frankvicky) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt:6]: `fun findByRoomId(roomId: String): Room?` -> `fun findByRoomId(roomId: Room.RoomId): Room?`

File: spring/src/main/kotlin/tw/waterballsa/gaas/spring/repositories/data/RoomData.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    gameId = gameId,
+                    hostId = hostId,
+                    playerIds = playerIds,
+                    maxPlayer = maxPlayer,
+                    minPlayer = minPlayer,
+                    name = name,
+                    description = description
+                )
+            }
+            room.roomId?.let { roomData.roomId = it.value }
```
Comment (frankvicky) [inline spring/src/main/kotlin/tw/waterballsa/gaas/spring/repositories/data/RoomData.kt:34]: 跟 24 行的邏輯似乎重複了？

File: spring/src/test/kotlin/tw/waterballsa/gaas/spring/controllers/RoomControllerTest.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@SpringBootTest
+@AutoConfigureMockMvc(addFilters = false)
+@ActiveProfiles(profiles = ["dev"])
+class RoomControllerTest {
+    @Autowired
+    lateinit var mockMvc: MockMvc
+
+    @Autowired
+    lateinit var roomRepository: RoomRepository
+
```
Comment (frankvicky) [inline spring/src/test/kotlin/tw/waterballsa/gaas/spring/controllers/RoomControllerTest.kt:21]: 要補上
```kotlin
@MockBean
lateinit var eventBus: EventBus
```

**Review by frankvicky — CHANGES_REQUESTED** [review]

File: application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package tw.waterballsa.gaas.application.repositories
+
+import tw.waterballsa.gaas.domain.Room
+
+interface RoomRepository {
+:    fun findByRoomId(roomId: Room.RoomId): Room?
```
Comment (frankvicky) [inline application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt:6]: 冒號？

**Review by frankvicky — CHANGES_REQUESTED** [review]

File: application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt (hunk 5 above)
Comment (cool9850311) [reply application/src/main/kotlin/tw/waterballsa/gaas/application/repositories/RoomRepository.kt]: 眼幹

**Review by frankvicky — APPROVED** [review]
LGTM

**Pull request merged**